### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by Daniel-Kigozi**
In this milestone, we did the following
-Add n+1
Screenshot from 2023-08-06 14-17-32
-After
Screenshot from 2023-08-06 14-43-29
- Add integration tests for views
image

### CODE REVIEW

File: spec/views/users/show.html.erb_spec.rb [hunk 1]
```diff
@@ -1,5 +1,60 @@
...
+    it 'has links to view individual posts' do
+      click_link(@post.title)
+      expect(page).to have_current_path(user_post_path(@user, @post))
+    end
+
+    it 'redirects to open all posts of a user' do
+      click_link('Show All')
+      expect(page).to have_current_path(user_posts_path(@user))
+    end
+  end
```
Comment (prolajumokeoni) [inline spec/views/users/show.html.erb_spec.rb]: - Please add this test in your users show page, it is currently missing 👍🏿 

- [ ] Tests that asserts you can see the number of posts the user has written.

File: spec/views/posts/index.html.erb_spec.rb [hunk 2]
```diff
@@ -1,5 +1,50 @@
...
+
+    it 'shows how many comments a post has.' do
+      expect(page).to have_content('Comments')
+    end
+
+    it 'shows how many likes a post has.' do
+      expect(page).to have_content('Likes')
+    end
+
+    it 'shows a section for pagination if there are more posts than fit on the view' do
```
Comment (prolajumokeoni) [inline spec/views/posts/index.html.erb_spec.rb]: - Please make sure tests written are passing, see screenshot below ⬇️ 
Screenshot_138

- Please add the following tests that are missing ⬇️ 

- [ ] Tests that asserts you can see the user's profile picture.
- [ ] Tests that redirects to the post's show page when you click on a post.

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Daniel-Kigozi and @rugwizangoga 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  solved N+1 problem when fetching all posts and their comments for a user




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @Daniel-Kigozi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All required tests added :heavy_check_mark: 
- All tests passing :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**